docs/DeveloperGuide.md

```diff
@@ -1,34 +1,972 @@
+
+> ℹ️ **Note:** The command <b>will not update grades</b> if the module requested has not been completed. The rest of 
+the information parsed in the command (if any) will be updated. 
+
+
+#### Details
+
+The following is the UML diagrams for update command consisting of updating each flag.
+
+![archi](./images/updateClassDiagram.jpg)
+
+<sup>***Figure 4.3.3.1** Class diagram of `Update` class with user input*</sup>
+
+
+![archi](./images/UpdateCommandSequenceDiagram.png)
+
+<sup>***Figure 4.3.3.2** Sequence diagram of `Update` class with user input*</sup>
+
+#### Considerations
+
+#### Alternative
+
+
+#### 4.3.4 List Command
+
+The list command provides users with 4 options to list down the modules being tracked by iGraduate. The 4 options are:
+1. List all modules being tracked:
+    - `all`
+1. List modules that have been marked as done:
+   - `complete`
+1. List modules that have not been marked as done:
+    - `incomplete`
+1. List incomplete modules available to be marked as done based on prerequisites completed:
+    - `available`
+    
+
+#### 4.3.5 CAP Command
+
+The CAP command calculates the current CAP of the user based on the grades of modules that are marked as done. The 
+command also displays the degree classification of the user. There are no flags or additional parameters required.
+
+![archi](./images/CapCommandSequenceDiagram.png)
+
+<sup>***Figure 4.3.5.1** Sequence diagram of `CapCommand` class.*</sup>
+
+#### 4.3.6 Done Command
+
+The done command is used to mark a module as completed. To execute this command, the module code is extracted as a 
+parameter from user input, and there is 1 compulsory flag:
+1. Grade obtained for module
+    - `-g <String>`
+
+> ℹ️ **Note:** Only NUS recognised grades are permitted for the grade flag.
+
+> ℹ️ **Note:** iGraduate recognises and supports either 'S' grade or 'U' grade.
+
+![archi](./images/DoneCommandSequenceDiagram.png)
+
+<sup>***Figure 4.3.6.1** Sequence diagram of `DoneCommand` class with user input*</sup>
+
+#### 4.3.7 Progress Command
+
+The progress command prints a progress bar as well as the user's graduation progress in the form of a percentage. No 
+additional flags are required for this command.
+
+![archi](./images/ProgressSequenceDiagram.png)
+
+<sup>***Figure 4.3.7.1** Sequence diagram of `ProgressCommand` class.*</sup>
+
+### 4.4 Module
+
+### 4.5 ModuleList
+
+### 4.6 Storage
+
+The storage feature saves the state of the module list after every execution of commands that manipulates 
+(i.e. update, add or delete) the modules in the list.
+
+#### Details
+
+The storage function is executed after every command that manipulates (i.e. adds, deletes or updates) the 
+modules in the module list, saving the updated state into the storage file. 
+The module list is stored in a storage file named `modules.json` in the `data` folder 
+(`<program location>/data/modules.json`). 
+
+![archi](./images/storageSequenceDiagram.jpg)
```

Comment:
Some of the diagrams (this included) appear different in format/style from the others. Perhaps it would be better to use a consistent format/style across the whole document?